src/main/scala/dsptools/numbers/chisel_types/UIntTypeClass.scala

```diff
@@ -34,7 +34,7 @@ trait UIntRing extends Any with Ring[UInt] with hasContext {
     }
     ShiftRegister(diff.asUInt, context.numAddPipes)
   }
-  def negate(f: UInt): UInt = -f
+  def negate(f: UInt): UInt = throw DspException("Can't negate UInt and get UInt")
```

Comment:
Should this be an exception? Minus is supported. Negation in the sense of additive inverse still makes sense for `UInt`s.

Reply:
Even though it's not representable as a UInt?